Source/Scene/HereMapsImageryProvider.js

```diff
@@ -0,0 +1,521 @@
+define([
+        '../Core/defaultValue',
+        '../Core/WebMercatorTilingScheme',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/Rectangle',
+        '../Scene/ImageryProvider',
+        '../Core/defined',
+        '../Core/Event',
+        '../Core/Math',
+        '../Core/Credit',
+        '../Core/TileProviderError',
+        '../Core/RuntimeError',
+        '../Core/Resource',
+        '../ThirdParty/when'
+    ], function(
+        defaultValue,
+        WebMercatorTilingScheme,
+        defineProperties,
+        DeveloperError,
+        Rectangle,
+        ImageryProvider,
+        defined,
+        CesiumEvent,
+        CesiumMath,
+        Credit,
+        TileProviderError,
+        RuntimeError,
+        Resource,
+        when) {
+    'use strict';
+
+    /**
+     * Provides tiled imagery using the Here Maps Imagery REST API. See https://developer.here.com/documentation/map-tile/topics/request-constructing.html
+     * for more details on how the query URLs are constructed and acceptable values for each option.
+     *
+     * @alias HereMapsImageryProvider.js
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} [options.baseUrl="aerial.maps.api.here.com"] The URL for accessing tiles, without the Load Balancing prefix.
```

Comment:
This parameter is usually name `url` on the other providers. It should also be a `{Resource|String}`. You can use `Resource.createIfNeeded` method to create or clone the resource.

Reply:
I did notice that's what the new ImageryProviders were doing. The problem is that HereMaps uses a subdomain [1,2,3, or 4] before all their requests. E.g. all urls start with http://2.aerial.... I couldn't see a good way of doing that with Resource, which seemed like it only wanted to handle Subresources (i.e. appending to the end of the url). That's also why I named it baseUrl, because there's no way of specifying a full url that would be useful. 

Should I just have options.url expect something like "http://aerial.maps.api.here.com" and then manually stick the subdomain in the middle there?